### PR TITLE
`torch.lobpcg.backward`: do not save non-Variable types with `ctx.save_for_backward`.

### LINKED ISSUE #67827: Can't backward through torch.lobpcg when using "largest" kwarg
## To Reproduce

With `torch.lobpcg`  in pytorch 1.9.1 .
```
torch.manual_seed(0)
a = torch.randn((100, 100), requires_grad=True)
a = a + a.T
torch.lobpcg(a, largest=False)
```
It will cause an error:
```
---------------------------------------------------------------------------

      1 a = torch.randn((100, 100), requires_grad=True)
      2 a = a + a.T
----> 3 torch.lobpcg(a, largest=False)
      4

~/Software/miniconda3/envs/python38/lib/python3.8/site-packages/torch/_lobpcg.py in lobpcg(A, k, B, X, n, iK, niter, tol, largest, method, tracker, ortho_iparams, ortho_fparams, ortho_bparams)
    518             B_sym = (B + B.transpose(-2, -1)) / 2 if (B is not None) else None
    519
--> 520             return LOBPCGAutogradFunction.apply(
    521                 A_sym, k, B_sym, X, n, iK, niter, tol, largest,
    522                 method, tracker, ortho_iparams, ortho_fparams, ortho_bparams

TypeError: save_for_backward can only save variables, but argument 4 is of type bool
```

It seems that I can not use `bool` for `largest`

## Environment
PyTorch version: 1.9.1         
Is debug build: False                
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A
                   
OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.21.3     
Libc version: glibc-2.27  
                                                                 
Python version: 3.8.11 (default, Aug  3 2021, 15:09:35)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.0-89-generic-x86_64-with-glibc2.17               
Is CUDA available: True                                                         
CUDA runtime version: 10.2.89

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7 @jianyuh @mruberry @walterddr @IvanYashchuk @xwang233

### PR DESCRIPTION
**Pull request by nikitaved**
Fixes https://github.com/pytorch/pytorch/issues/67827

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7

### CODE REVIEW

**Review by lezcano — APPROVED** [review]
Thanks for the clean-up of the test as well!

File: test/test_autograd.py [hunk 1]
```diff
@@ -2697,7 +2697,6 @@ def test_var_mean_differentiable(self):
         torch.autograd.backward(r2, grad)
         self.assertEqual(input1.grad, input2.grad, rtol=0.01, atol=0.0)
 
-    @slowTest
```
Comment (albanD) [inline test/test_autograd.py:2700]: Not slow anymore? This was forgotten in the gradcheck speed up?

**Review by albanD — APPROVED** [review]
Thanks!

File: test/test_autograd.py (hunk 1 above)
Comment (nikitaved) [reply test/test_autograd.py:2700]: It takes 3 seconds in my environment, not 1-2 minutes from long ago.

Comment (albanD) [reply test/test_autograd.py:2700]: Awesome!

**Review by soulitzer — APPROVED** [review]

**Pull request closed without merging**